src/algebraic_topology/simplex_category.lean

```diff
@@ -494,6 +494,257 @@ end
+end
+
+lemma bijective_of_mono_and_eq {x y : simplex_category.{u}} (i : x ⟶ y) [mono i]
+  (hxy : x = y) : function.bijective i.to_order_hom :=
+by simpa only [fintype.bijective_iff_injective_and_card i.to_order_hom,
+    ← mono_iff_injective, hxy, and_true, eq_self_iff_true]
+
+lemma bijective_of_epi_and_eq {x y : simplex_category.{u}} (e : x ⟶ y) [epi e]
+  (hxy : x = y) : function.bijective e.to_order_hom :=
+by simpa only [fintype.bijective_iff_surjective_and_card e.to_order_hom,
+    ← epi_iff_surjective, hxy, and_true, eq_self_iff_true]
+
+/-- A bijective map in `simplex_category` is an isomorphism. -/
+@[simps]
+noncomputable def iso_of_bijective {x y : simplex_category.{u}} {f : x ⟶ y}
+  (hf : function.bijective (f.to_order_hom.to_fun)) : x ≅ y :=
+{ hom := f,
+  inv := hom.mk
+    { to_fun := function.inv_fun f.to_order_hom.to_fun,
+      monotone' := λ y₁ y₂ h, begin
+        by_cases h' : y₁ < y₂,
+        { by_contradiction h'',
+          have ineq := f.to_order_hom.monotone' (le_of_not_ge h''),
+          have eq := λ i, function.inv_fun_eq (function.bijective.surjective hf i),
+          simp only at eq,
+          simp only [eq] at ineq,
+          exact not_le.mpr h' ineq, },
+        { rw eq_of_le_of_not_lt h h', }
+      end, },
+  hom_inv_id' := begin
+    ext1, ext1, ext1 i,
+    apply function.left_inverse_inv_fun (function.bijective.injective hf),
+  end,
+  inv_hom_id' := begin
+    ext1, ext1, ext1 i,
+    apply function.right_inverse_inv_fun (function.bijective.surjective hf),
+  end, }
+
+lemma is_iso_of_bijective {x y : simplex_category.{u}} {f : x ⟶ y}
+  (hf : function.bijective (f.to_order_hom.to_fun)) : is_iso f :=
+begin
+  rw [show f = (iso_of_bijective hf).hom, by simp only [iso_of_bijective_hom]],
+  apply_instance,
+end
+
+/-- An isomorphism in `simplex_category` induces an `order_iso`. -/
+@[simp]
+def order_iso_of_iso {x y : simplex_category.{u}} (e : x ≅ y) :
+  fin (x.len+1) ≃o fin (y.len+1) :=
+equiv.to_order_iso
+  { to_fun    := e.hom.to_order_hom,
+    inv_fun   := e.inv.to_order_hom,
+    left_inv  := λ i, by simpa only using congr_arg (λ φ, (hom.to_order_hom φ) i) e.hom_inv_id',
+    right_inv := λ i, by simpa only using congr_arg (λ φ, (hom.to_order_hom φ) i) e.inv_hom_id', }
+  e.hom.to_order_hom.monotone e.inv.to_order_hom.monotone
+
+lemma iso_eq_iso_refl {x : simplex_category.{u}} (e : x ≅ x) :
+  e = iso.refl x :=
+begin
+  have h : (finset.univ : finset (fin (x.len+1))).card = x.len+1 := finset.card_fin (x.len+1),
+  have eq₁ := finset.order_emb_of_fin_unique' h
+    (λ i, finset.mem_univ ((order_iso_of_iso e) i)),
+  have eq₂ := finset.order_emb_of_fin_unique' h
+    (λ i, finset.mem_univ ((order_iso_of_iso (iso.refl x)) i)),
+  ext1, ext1,
+  convert congr_arg (λ φ, (order_embedding.to_order_hom φ)) (eq₁.trans eq₂.symm),
+  ext1, ext1 i,
+  refl,
+end
+
+lemma eq_eq_to_iso_of_iso {x y : simplex_category.{u}} (e : x ≅ y) :
```

Comment:
I don't understand what this lemma and the next are for. Seems like you could obtain `eq_id_of_is_iso` in a more transparent way.